invenio_communities/assets/semantic-ui/js/invenio_communities/community/searchComponents/CommunitySelectionSearch.js

```diff
@@ -0,0 +1,252 @@
+// This file is part of Invenio-RDM-Records
+// Copyright (C) 2024 CERN.
+//
+// Invenio-communities is free software; you can redistribute it and/or modify it
+// under the terms of the MIT License; see LICENSE file for more details.
+
+import { i18next } from "@translations/invenio_rdm_records/i18next";
+import React, { Component } from "react";
+import { OverridableContext } from "react-overridable";
+import { InvenioSearchApi, ReactSearchKit, SearchBar, buildUID } from "react-searchkit";
+import { Button, Grid, Icon, Menu } from "semantic-ui-react";
+import PropTypes from "prop-types";
+import {
+  SearchConfigurationContext,
+  SearchAppFacets,
+  SearchAppResultsPane,
+} from "@js/invenio_search_ui/components";
+import { GridResponsiveSidebarColumn, InvenioPopup } from "react-invenio-forms";
+
+export class CommunitySelectionSearch extends Component {
+  constructor(props) {
+    super(props);
+    const {
+      apiConfigs: { allCommunities },
+    } = this.props;
+
+    this.state = {
+      selectedConfig: allCommunities,
+      sidebarVisible: false,
+    };
+  }
+
+  prefixAppID(components, appID) {
```

Comment:
Needed for overridable components. The `appID` is related to the state of the component, based on the selected config (all communities X my communities)

Reply:
could you leave a comment describing in a bit more details on why it is needed?
example id built by this function would be nice to reference in this comment

Reply:
> could you leave a comment describing in a bit more details on why it is needed? example id built by this function would be nice to reference in this comment

Absolutely, will do!